sonar-orchestrator/src/main/java/com/sonar/orchestrator/container/Server.java

```diff
@@ -173,100 +138,79 @@ private File getLogFile(String logFile) {
     return FileUtils.getFile(home, "logs", logFile);
   }
 
+  /**
+   * @deprecated in 3.15. Instantiate your own client configured with {@link #getUrl()}
+   * or use {@link #newHttpCall(String)}
+   */
+  @Deprecated
   public Sonar getWsClient() {
     if (wsClient == null && url != null) {
-      wsClient = new Sonar(new HttpClient4Connector(new Host(url)));
+      wsClient = new Sonar(new HttpClient4Connector(new Host(url.toString())));
     }
     return wsClient;
   }
 
   /**
    * @since 2.10
+   * @deprecated in 3.15. Instantiate your own client configured with {@link #getUrl()}
+   * or use {@link #newHttpCall(String)}
    */
+  @Deprecated
   public SonarClient wsClient() {
     if (sonarClient == null && url != null) {
-      sonarClient = SonarClient.create(url);
+      sonarClient = SonarClient.create(url.toString());
     }
     return sonarClient;
   }
 
   /**
    * @since 2.10
+   * @deprecated in 3.15. Instantiate your own client configured with {@link #getUrl()}
+   * or use {@link #newHttpCall(String)}
    */
+  @Deprecated
   public SonarClient wsClient(String login, String password) {
-    return SonarClient.builder().url(url).login(login).password(password).build();
+    return SonarClient.builder().url(url.toString()).login(login).password(password).build();
   }
 
+  /**
+   * @deprecated in 3.15. Instantiate your own client configured with {@link #getUrl()}
+   * or use {@link #newHttpCall(String)}
+   */
+  @Deprecated
   public Sonar getAdminWsClient() {
     if (adminWsClient == null && url != null) {
-      adminWsClient = new Sonar(new HttpClient4Connector(new Host(url, ADMIN_LOGIN, ADMIN_PASSWORD)));
+      adminWsClient = new Sonar(new HttpClient4Connector(new Host(url.toString(), ADMIN_LOGIN, ADMIN_PASSWORD)));
     }
     return adminWsClient;
   }
 
+  /**
+   * @deprecated in 3.15. Instantiate your own client configured with {@link #getUrl()}
+   * or use {@link #newHttpCall(String)}
+   */
+  @Deprecated
   public SonarClient adminWsClient() {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)